sycl/include/CL/sycl/handler.hpp

```diff
@@ -1560,12 +1567,20 @@ class __SYCL_EXPORT handler {
     verifyUsedKernelBundle(detail::KernelInfo<NameT>::getName());
     using LambdaArgType =
         sycl::detail::lambda_arg_type<KernelType, nd_item<Dims>>;
+    // If type is convertible to sycl::item/sycl::nd_item, convert it to
+    // sycl::item/sycl::nd_item
+    using TransformedArgType = typename std::conditional<
+        std::is_convertible<nd_item<Dims>, LambdaArgType>::value, nd_item<Dims>,
+        typename std::conditional<
+            std::is_convertible<item<Dims>, LambdaArgType>::value, item<Dims>,
+            LambdaArgType>::type>::type;
```

Comment:
It looks like the same code is used in two places. Could you move it to the helper to be called in both cases?

Reply:
Extracted to separate helper struct